src/mlpack/methods/kde/kde_impl.hpp

```diff
@@ -749,6 +771,179 @@ serialize(Archive& ar, const unsigned int version)
   ar & BOOST_SERIALIZATION_NVP(oldFromNewReferences);
 }
 
+template<typename KernelType,
+         typename MetricType,
+         typename MatType,
+         template<typename TreeMetricType,
+                  typename TreeStatType,
+                  typename TreeMatType> class TreeType,
+         template<typename> class DualTreeTraversalType,
+         template<typename> class SingleTreeTraversalType>
+void KDE<KernelType,
+         MetricType,
+         MatType,
+         TreeType,
+         DualTreeTraversalType,
+         SingleTreeTraversalType>::
+ComputePCA(Tree& rootNode)
+{
+  Timer::Start("computing_dimensionality_reduction");
+  std::stack<Tree*> nodes;
+  KDEStackRules<Tree> reverseRules(nodes);
+  SingleTreeTraversalType<KDEStackRules<Tree>> reverseTraverser(reverseRules);
+  reverseTraverser.Traverse(0, rootNode);
+
+  while (!nodes.empty())
+  {
+    Tree* node = nodes.top();
+    nodes.pop();
+    KDEStat& stat = node->Stat();
+
+    // Compute base.
+    if (node->IsLeaf())
+    {
+      const size_t numPoints = node->NumPoints();
+      arma::mat pointsInTheNode(node->Dataset().n_rows, numPoints);
+      for (size_t i = 0; i < numPoints; ++i)
+        pointsInTheNode.col(i) = node->Dataset().col(node->Point(i));
+
+      // Amount of points for which the base has been calculated.
+      stat.PointsInTheBase() = numPoints;
+
+      // Calculate base mean.
+      stat.Mean() = arma::mean(pointsInTheNode, 1);
+
+      // Right singular values. Unused.
+      arma::mat v;
+
+      // Singular value decomposition.
+      if (pointsInTheNode.n_rows > pointsInTheNode.n_cols)
+        arma::svd_econ(stat.EigVec(), stat.EigVal(), v, pointsInTheNode, "l");
+      else
+        arma::svd(stat.EigVec(), stat.EigVal(), v, pointsInTheNode);
```

Comment:
Ok---I see the algorithm you are using here is SVD (not the eigendecomposition) and as far as I can tell this does correctly recover the eigenvectors for the covariance matrix.